File: Problem/By section/Greedy/Problem 455/problem_455.java
import java.util.*;

class Solution {
    // Time: O(nlog(n)) or O(n + m), Space : O(1)
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        
        // two pointer
        int i = 0;
        int j = 0;
        int ans = 0;
        
        while(i < g.length && j < s.length){
            // we will assign the current least size cookie to the children will least content value
            // if feasible
            if(s[j] >= g[i]){
                ans++;
                i++;
                j++;
            }else{
                j ++;
            }
        }
        
        return ans;
    }
}

File: Problem/By section/Greedy/Problem 455/problem_455.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    // Time: O(nlogn), Space: O(1)
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());

        int j = 0, k = 0;
        int answer = 0;
        while(j < g.size() && k < s.size()) {
            if(g[j] <= s[k]) {
                answer++;
                j++;
                k++;
            }else{
                k++;
            }
        }

        return answer;
    }
};

int main(int argc, char** argv) {
    return 0;
}